src/sentry/snuba/metrics/extraction.py

```diff
@@ -700,7 +708,7 @@ def should_use_on_demand_metrics(
                 prefilling=prefilling,
             )
             metrics.incr("on_demand_metrics.should_use_on_demand_metrics.cache_miss")
-            organization_bulk_query_cache[local_cache_key] = result
+            organization_bulk_query_cache[local_cache_digest_chunk][local_cache_key] = result
```

Comment:
```suggestion
            organization_bulk_query_cache[local_cache_digest_chunk].get(local_cache_key, False) = result
```
I think `local_cache_key` will still keyerror here cause the left side will give you `{}` then you're doing `{}[local_cache_key]`